### PR TITLE
Query: Skip formatting strings if debug logging is disabled

### PR DESCRIPTION
**Pull request by wallee94**
After upgrading from v0.34.1 to v0.36.0, I see some queriers using about 10 times more CPU than before. These queriers only receive ~16 instant http q/s and have over 20 endpoints.

A pprof profile showed that the GC is using most of the CPU, so I ran a pprof heap alloc_space, and the problem seems to be `fmt.Sprintf` used in the debug message in the store proxy, even if debug logging is disabled.

image

* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes

I added a validation to skip the string formatting if debug logging is disabled.

## Verification

I deployed a patch with this change and it definitely fixed the CPU usage. This graph is avg CPU usage after removing the formatting in `matchingStores` and then after removing it in `storeMatches` and `storeMatchDebugMetadata`.

image

### CODE REVIEW

**Review by yeya24 — APPROVED** [review]
Thanks. This LGTM. Thanks for fixing the issue

**Comment by MichaHoffmann** [comment]
Do we actually need those debugging messages at all?

**Review by saswatamcode — APPROVED** [review]

File: pkg/store/proxy.go [hunk 1]
```diff
@@ -591,22 +597,30 @@ func storeMatches(ctx context.Context, s Client, mint, maxt int64, matchers ...*
 
 	storeMinTime, storeMaxTime := s.TimeRange()
 	if mint > storeMaxTime || maxt < storeMinTime {
-		return false, fmt.Sprintf("does not have data within this time period: [%v,%v]. Store time ranges: [%v,%v]", mint, maxt, storeMinTime, storeMaxTime)
+		s := "does not have data within this time period"
```
Comment (yeya24) [inline pkg/store/proxy.go:600]: We can just make it a const

Comment (wallee94) [reply pkg/store/proxy.go:600]: Sure! You mean inside the function `const s string = "..."`, or package-level?

Comment (pedro-stanaka) [reply pkg/store/proxy.go:600]: I think it should be fine as local const. Since we are not using anywhere else.

Comment (wallee94) [reply pkg/store/proxy.go]: Done ✅

**Review by saswatamcode — APPROVED** [review]
LGTM

**Pull request merged**